filament/src/IndexBuffer.cpp

```diff
@@ -58,30 +59,42 @@ IndexBuffer* IndexBuffer::Builder::build(Engine& engine) {
 // ------------------------------------------------------------------------------------------------
 
 FIndexBuffer::FIndexBuffer(FEngine& engine, const IndexBuffer::Builder& builder)
-        : mIndexCount(builder->mIndexCount),
-          mExternalBuffersEnabled(builder->mExternalBuffersEnabled) {
+        : mIndexCount(builder->mIndexCount), mBufferObjectEnabled(builder->mBufferObjectEnabled) {
     FEngine::DriverApi& driver = engine.getDriverApi();
     mHandle = driver.createIndexBuffer(
             (backend::ElementType)builder->mIndexType,
             uint32_t(builder->mIndexCount),
-            backend::BufferUsage::STATIC, 
-            mExternalBuffersEnabled);
+            backend::BufferUsage::STATIC);
+
+    // If buffer objects are not enabled at the API level, then we create them internally.
+    if (!mBufferObjectEnabled) {
+        mObjectHandle = engine.getDriverApi().createBufferObject(
+            mIndexCount * ((builder->mIndexType == IndexType::UINT) ? 4U : 2U),
```

Comment:
nit: Would it be worth introducing a `Driver::getIndexTypeSize()`, similar to `getElementTypeSize()`?

Reply:
`Driver` doesn't know about the `IndexType` enum, because it is defined in the `IndexBuffer` class. I would change this line to  `Driver::getElementTypeSize((backend::ElementType)builder->mIndexType)` because that's how the driver do it in the backend.

Reply:
fixed in https://github.com/shapr3d/filament/pull/89/commits/062921c653c8f3116da92e6a3e8cba729d402d59